java/vector/src/main/java/org/apache/arrow/vector/BaseLargeVariableWidthVector.java

```diff
@@ -73,7 +73,8 @@ public BaseLargeVariableWidthVector(Field field, final BufferAllocator allocator
     lastValueCapacity = INITIAL_VALUE_ALLOCATION - 1;
     valueCount = 0;
     lastSet = -1;
-    offsetBuffer = allocator.getEmpty();
+    // According to Arrow spec, the offsets buffer contains length + 1 elements
+    allocateOffsetBuffer(OFFSET_WIDTH);
```

Comment:
It might be expensive to do this up front every time an empty vector is created. Should this be done lazily?

Reply:
Hmm, maybe I should update the export code of C data interface here? I think previously the empty offset buffer works like a lazy offset buffer in var-wide arrays. They are expanded to valid format (length + 1) once the array has elements.